File: BespokeJokes.Server/Models/Joke.cs
using System.ComponentModel.DataAnnotations;

namespace BespokeJokes.Server.Models
{
    public class Joke
    {
        [Key]
        public int JokeId { get; set; }
        public string JokeText { get; set; }
        public string PunchLineText { get; set; }
        public string Author { get; set; }
        public DateTime PostDate { get; set; }
        public DateTime UpdateDate { get; set; }

    }
}

File: bespokejokes.client/src/app/Models/Joke.ts
export interface Joke {

  jokeId: number,
  jokeText: string,
  punchLineText: string,
  author: string,
  postDate: Date,
  updateDate: Date
}
export const emptyJoke: Joke = {
    jokeId: 0,
    jokeText: "",
    punchLineText: "",
    author: "",
  postDate: new Date("2025-03-15"),
  updateDate: new Date("2025-03-15")
}
